### DISCUSSION TITLE
Global state

### DISCUSSION THREAD

**Discussion by chanan** [rrousselGit/riverpod#2143] [Q&A] [answered]
Hello,

I am just learning Riverpod, so this might be a basic question, but I could not find the answer to it.

I am using v2 of Riverpod. I have an `AsyncNotifierProvider` that the state contains a user Account. The provider logins the user in and loads the account from the database. Later in the application I need the account do some operation, but when I watch it, the state has been reset. I assume it's do to the auto dispose. How can I have an AsyncNotifierProvider that does not dispose / make sure it doesn't dispose?

**Comment by rrousselGit** [comment, marked as answer]
Either `ref.watch` in certain widgets to keep the provider alive, or remove `autoDispose`

**Comment by chanan** [comment]
1. Do you mean for example in a top parent of my app?
2. How do I remove the auto dispose when using `@riverpod`?

**Reply by riscait** [reply]
> How do I remove the auto dispose when using @riverpod?

You can use `@Riverpod(keepAlive: true)`.
